File: web-api/src/main/java/cn/labzen/web/api/annotation/Crypto.java
package cn.labzen.web.api.annotation;

import java.lang.annotation.*;

/**
 * API 响应数据加密注解。
 * <p>
 * 注解在 Controller 接口或方法上，对 API 响应数据进行加密处理。
 * <p>
 * <b>待实现：</b> 定义加密算法、密钥配置等参数。
 */
@Documented
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface Crypto {
}

File: src/main/kotlin/cn/labzen/web/annotation/Crypto.kt
package cn.labzen.web.annotation

/**
 * 注解在 Controller 接口或方法上，对API响应数据进行加密
 *
 * todo 待实现
 */
@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.SOURCE)
annotation class Crypto()
